flutter-idea/src/io/flutter/FlutterInitializer.java

```diff
@@ -255,6 +276,83 @@ private void setUpDtdAnalytics(Project project) {
     t1.start();
   }
 
+  private void setUpThemeChangeNotifications(Project project) {
+    if (project == null) return;
+    FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
+    if (sdk == null || !sdk.getVersion().canUseDtd()) return;
+    Thread t1 = new Thread(() -> {
+      if (busSubscribed) return;
+      final MessageBusConnection connection = ApplicationManager.getApplication().getMessageBus().connect();
+      connection.subscribe(EditorColorsManager.TOPIC, (EditorColorsListener)scheme -> {
+        sendThemeChangedEvent(project);
+      });
+      connection.subscribe(UISettingsListener.TOPIC, (UISettingsListener)scheme -> {
+        sendThemeChangedEvent(project);
+      });
+      busSubscribed = true;
+    });
+    t1.start();
+  }
+
+  private void sendThemeChangedEvent(@NotNull Project project) {
+    // Debounce this request because the topic subscriptions can trigger multiple times (potentially from initial notification of change and
+    // also from application of change)
+
+    // Set the current time of this request
+    final long requestTime = System.currentTimeMillis();
+    lastScheduledThemeChangeTime.set(requestTime);
+
+    // Schedule event to be sent in a second if nothing more recent has come in.
+    Executors.newSingleThreadScheduledExecutor().schedule(() -> {
+      if (lastScheduledThemeChangeTime.get() != requestTime) {
+        System.out.println("this request is being dropped:  " + requestTime);
+        System.out.println("last scheduled time is: " + lastScheduledThemeChangeTime.get());
+        // A more recent request has been set, so drop this request.
+        return;
+      }
+
+      System.out.println("this request is being sent:  " + requestTime);
+      final JsonObject params = new JsonObject();
+      params.addProperty("eventKind", "themeChanged");
```

Comment:
@DanTup do these fields follow the same conventions as editor events that you're sending from VS Code?

Reply:
At a glance, this looks the same to me :)

My code is here:

https://github.com/Dart-Code/Dart-Code/blob/1dfe247fb5a4d0fbed25283d80067c5ce77ee286/src/shared/services/tooling_daemon.ts#L190-L198

Reply:
Okay great - mainly I was concerned because I couldn't tell from your code whether "themeChanged" should be uppercase etc.

Reply:
Ah, that comes from the enum (and your casing is correct):

https://github.com/Dart-Code/Dart-Code/blob/1dfe247fb5a4d0fbed25283d80067c5ce77ee286/src/shared/services/tooling_daemon_services.ts#L150

Reply:
thanks!